### PR TITLE
User authentication for both /configure and /keyboards

### LINKED ISSUE #388: Login feature for /configure
We need to provide a user authentication (that is, login) feature to implement some other features including a new feature to share keymaps with other users.

### PR DESCRIPTION
**Pull request by yoichiro**
Fix #388 

右上のアイコンをクリックした後に出てくるメニューで「Login」を選択した後

スクリーンショット 2021-03-12 6 28 11

１枚めのダイアログのGoogle Accountをクリックした後

スクリーンショット 2021-03-12 6 29 09

認証後の画面

スクリーンショット 2021-03-12 6 29 27

If a user who has already been signed in with Google account accesses to `/keyboards`, Remap asks an authentication to  link a GitHub account to the Google account.

### CODE REVIEW

File: src/components/configure/header/Header.tsx [hunk 1]
```diff
@@ -97,6 +110,159 @@ export default class Header extends React.Component<HeaderProps, HeaderState> {
...
+          <Menu
+            id="configure-header-auth-menu"
+            anchorEl={authMenuAnchorEl}
+            open={Boolean(authMenuAnchorEl)}
+            onClose={this.handleAuthMenuClose}
+          >
+            {this.renderLinkGoogleAccountMenu()}
+            {this.renderLinkGitHubAccountMenu()}
+            <MenuItem
+              key="3"
```
Comment (adamrocker) [inline src/components/configure/header/Header.tsx:221]: It's not problem in this case, for your reference, using the index as a key might be unti-pattern in React. https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

File: src/components/configure/header/Header.tsx [hunk 2]
```diff
@@ -97,6 +110,159 @@ export default class Header extends React.Component<HeaderProps, HeaderState> {
...
+    this.setState({ authMenuAnchorEl: null });
+    this.props.linkToGoogleAccount!();
+  }
+
+  async handleLinkGitHubAccountMenuClick() {
+    this.setState({ authMenuAnchorEl: null });
+    this.props.linkToGitHubAccount!();
+  }
+
+  renderLinkGoogleAccountMenu() {
```
Comment (adamrocker) [inline src/components/configure/header/Header.tsx:146]: FYI: JS's top-revel function which return jsx is useful in this case.
I believe a good sample is as below:
https://github.com/remap-keys/remap/blob/main/src/components/configure/keymap/Keymap.tsx#L306-L328

**Comment by adamrocker** [comment]
@yoichiro LGTM!!

File: src/components/configure/header/Header.tsx (hunk 1 above)
Comment (yoichiro) [reply src/components/configure/header/Header.tsx]: @adamrocker Thank you for the comment! Yes, each value specified by me was such index value. I changed the values at dc1b1fa.

File: src/components/configure/header/Header.tsx (hunk 2 above)
Comment (yoichiro) [reply src/components/configure/header/Header.tsx:146]: @adamrocker I could not decide which creating top-level function or creating render method is better until now. Ok, I intend to create such top-level function from next opportunity.

**Comment by yoichiro** [comment]
@adamrocker Thank you for the reviewing!

**Pull request merged**